### PR TITLE
[New Extension] Media Key Emulate

### PR DESCRIPTION
**Pull request by douo**
## Description

Emulate keyboard media keys press. Reassign hotkey for media keys. This is particularly useful for those who use a 71-key keyboard.

## Screencast

global-media-key-1

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by mil3na** [comment]
Hi! Thank you so much for your contribution to Raycast!
My name is Milena and I will be reviewing your extension today :)

We have an extension at the Store that allows you to emulate any `Keyboard Shortcut Sequences` here: https://www.raycast.com/HelloImSteven/keyboard-shortcut-sequences

Is there anything that your extension brings that is not achievable with the extension mentioned before?

**Comment by douo** [comment]
@mil3na Thank you for letting me know.`Keyboard Shortcut Sequences` is a great extension, but it uses quicklink to implement global hotkey for custom key sequences. Due to the limitations of raycast, there will be a brief flash of the raycast interface when pressing the hotkey, which is unacceptable to me. `Media Key Emulate` is aim to seamless remap media keys

https://github.com/raycast/extensions/assets/743074/3c66b3ee-8109-4951-a50c-0cf90cf4ca22

File: extensions/global-media-key/README-zh.md
Comment (mil3na) [inline extensions/global-media-key/README-zh.md:1]: Do you mind renaming this file to just` README.md` ?

File: extensions/global-media-key/package.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "global-media-key",
+  "title": "Media Key Emulate",
+  "description": "Emulate keyboard media keys press. Reassign hotkey for media keys.",
+  "icon": "extension_icon.png",
+  "author": "douo",
+  "categories": [
+    "System"
```
Comment (mil3na) [inline extensions/global-media-key/package.json:9]: Maybe you should add `Media` as a category too. What do you think?

File: extensions/global-media-key/package.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    {
+      "name": "rewind",
+      "description": "Toggle Play/Pause",
+      "title": "Rewind",
+      "icon": "rewind.png",
+      "mode": "no-view"
+    }
+  ],
+  "dependencies": {
+    "@raycast/api": "^1.48.9"
```
Comment (mil3na) [inline extensions/global-media-key/package.json:50]: Do you mind updating to the latest API version?

File: extensions/global-media-key/src/base.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    await fs.promises.access(binary, fs.constants.X_OK);
+  } catch {
+    await fs.promises.chmod(binary, 0o775);
+  }
+  showHUD(`send Key: ${type}`);
+  _exec(binary, type);
+}
+
+function _exec(binary: string, type: string) {
+  // const { status, output, stdout, stderr, error } = spawnSync(binary, [type], {
```
Comment (mil3na) [inline extensions/global-media-key/src/base.ts:19]: If this is not used, maybe can be removed.

File: extensions/global-media-key/package.json [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      "name": "previous",
+      "description": "Toggle Play/Pause",
+      "title": "Previous",
+      "icon": "previous.png",
+      "mode": "no-view"
+    },
+    {
+      "name": "fast",
+      "description": "Toggle Play/Pause",
+      "title": "Fast",
```
Comment (mil3na) [inline extensions/global-media-key/package.json:37]: Maybe use '`Forward`' instead of `Fast`?

**Review by mil3na — CHANGES_REQUESTED** [review]
That's a fair point. As this is something we cannot fix now, I think it's ok to let it in. 

Testing your extension, I noticed that the `Previous` command has the same behavior as `Rewind`. `Next` and `Fast` do have different behaviors. Is this a bug?

File: extensions/global-media-key/README-zh.md
Comment (douo) [reply extensions/global-media-key/README-zh.md:1]: My fault, I forgot to update the auto commit script.

File: extensions/global-media-key/package.json (hunk 1 above)
Comment (douo) [reply extensions/global-media-key/package.json:9]: Sure!

File: extensions/global-media-key/package.json (hunk 2 above)
Comment (douo) [reply extensions/global-media-key/package.json:50]: Sure!

**Comment by douo** [comment]
> That's a fair point. As this is something we cannot fix now, I think it's ok to let it in.
> 
> Testing your extension, I noticed that the `Previous` command has the same behavior as `Rewind`. `Next` and `Fast` do have different behaviors. Is this a bug?

Hi @mil3na, this extension only emulates the key event. The actual behavior is controlled by the software that consumes this key event.  You can verify the key code through this page: https://opensource.apple.com/source/IOHIDFamily/IOHIDFamily-34.12/IOHIDSystem/IOKit/hidsystem/ev_keymap.h.auto.html#:~:text=%23define%20NX_KEYTYPE_REWIND%09%0920

**Comment by mil3na** [comment]
Got it. Thanks for making the changes! Looks good to me :)

**Review by mil3na — APPROVED** [review]

**Pull request merged**